File: java/src/com/nrkei/microservices/car_rental_offer/Monitor.java
package com.nrkei.microservices.car_rental_offer;
/*
 * Copyright (c) 2016 by Fred George
 * May be used freely except for training; license required for training.
 * @author Fred George
 */

import com.nrkei.microservices.rapids_rivers.Packet;
import com.nrkei.microservices.rapids_rivers.PacketProblems;
import com.nrkei.microservices.rapids_rivers.RapidsConnection;
import com.nrkei.microservices.rapids_rivers.River;
import com.nrkei.microservices.rapids_rivers.rabbit_mq.RabbitMqRapids;

// Understands the messages on an event bus
public class Monitor implements River.PacketListener {

    public static void main(String[] args) {
        String host = args[0];
        String port = args[1];

        final RapidsConnection rapidsConnection = new RabbitMqRapids("monitor_all", host, port);
        final River river = new River(rapidsConnection);
        river.register(new Monitor());
        rapidsConnection.register(river);
    }

    @Override
    public void packet(RapidsConnection connection, Packet packet, PacketProblems warnings) {
        System.out.println(String.format(" [>] %s", warnings));
    }

    @Override
    public void onError(RapidsConnection connection, PacketProblems errors) {
        System.out.println(String.format(" [>] %s", errors));
    }
}

File: ruby_v2/lib/rental_offer/monitor.rb
#!/usr/bin/env ruby
# encoding: utf-8

# Copyright (c) 2017 by Fred George.
# May be used freely except for training; license required for training.

require 'securerandom'
require 'pry'

require '../rapids_rivers/rabbit_mq/rabbit_mq_rapids'

# Understands the complete stream of messages on an event bus
class Monitor
  attr_reader :service_name

  def initialize(host_ip, port)
    rapids_connection = RabbitMqRapids.new(host_ip, port)
    @river = River.new(rapids_connection)
    @service_name = 'monitor_ruby_' + SecureRandom.uuid
  end

  def start
    @river.register(this)
    puts " [*] Waiting for traffic on RabbitMQ event bus ... To exit press CTRL+C"
  end

  def packet rapids_connection, packet, warnings
    puts " [*] #{warnings}"
  end

  def on_error rapids_connection, errors
    puts " [*] #{errors}"
  end

end

Monitor.new(ARGV.shift, ARGV.shift).start
